src/Flysystem/FlysystemAssetStore.php

```diff
@@ -913,24 +954,94 @@ public static function flush()
 
     public function getResponseFor($asset)
     {
-        // Check if file exists
-        $filesystem = $this->getFilesystemFor($asset);
-        if (!$filesystem) {
-            return $this->createMissingResponse();
+
+        $public = $this->getPublicFilesystem();
+        $protected = $this->getProtectedFilesystem();
+
+        // If the file exists on the public store, we just straight return it.
+        if ($public->has($asset)) {
+            return $this->createResponseFor($public, $asset);
         }
 
-        // Block directory access
-        if ($filesystem->get($asset) instanceof Directory) {
```

Comment:
Why is this check no longer required? If you look at the implementation of `Local->has()`, it just does a `file_exists()`, which returns true for both files and folders.

Reply:
The check has been moved to `createResponseFor`? With the way I've reorganised `getResponseFor()` it made a bit more sense to have it there.

Reply:
True, so it doesn't have to get the asset twice from the asset store (once in `getResponseFor()`, then again in `createResponseFor()`. It does mean that it'll check `isGranted()` for protected folders though. I guess that's pretty low impact.